File: visual-model-ui/src/main/java/org/visual/model/ui/RoundedButton.java
package org.visual.model.ui;

import javafx.geometry.Insets;
import javafx.scene.control.Button;
import javafx.scene.control.Label;
import javafx.scene.layout.StackPane;
import javafx.scene.paint.Color;
import javafx.scene.shape.Circle;
import lombok.Getter;
import lombok.Setter;
import lombok.val;
import org.jetbrains.annotations.NotNull;

@Getter
@Setter
public class RoundedButton extends Button {

    private final Circle circle;

    private final Label label;

    public RoundedButton(String buttonText) {
        super();
        this.circle = createCircle();
        this.label = createLabel(buttonText);
        initialize();
    }

    private @NotNull Circle createCircle() {
        val newCircle = new Circle();
        newCircle.setRadius(10);
        newCircle.setFill(Color.LIGHTGRAY);
        newCircle.setStroke(Color.BLACK);
        return newCircle;
    }

    private @NotNull Label createLabel(String text) {
        Label newLabel = new Label(text);
        newLabel.setTextFill(Color.BLACK);
        return newLabel;
    }

    private void initialize() {
        val stackPane = new StackPane(circle, label);
        setGraphic(stackPane);
        setPadding(new Insets(5));
    }
}

File: visual-model-ui/src/main/kotlin/org/visual/model/ui/control/RoundedButton.kt
package org.visual.model.ui.control

import javafx.geometry.Insets
import javafx.scene.control.Button
import javafx.scene.control.Label
import javafx.scene.layout.StackPane
import javafx.scene.paint.Color
import javafx.scene.shape.Circle
import lombok.Getter
import lombok.Setter

@Getter
@Setter
class RoundedButton(buttonText: String) : Button() {
    private val circle: Circle

    private val label: Label

    init {
        this.circle = createCircle()
        this.label = createLabel(buttonText)
    }

    init {
        val stackPane = StackPane(circle, label)
        graphic = stackPane
        padding = Insets(5.0)
    }

    private fun createCircle(): Circle {
        val newCircle = Circle()
        newCircle.radius = 10.0
        newCircle.fill = Color.LIGHTGRAY
        newCircle.stroke = Color.BLACK
        return newCircle
    }

    private fun createLabel(text: String): Label {
        val newLabel = Label(text)
        newLabel.textFill = Color.BLACK
        return newLabel
    }
}
